pkg/apis/core/v1alpha1/types_collectedstatus.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+This file may have been modified by The KubeAdmiral Authors
+("KubeAdmiral Modifications"). All KubeAdmiral Modifications
+are Copyright 2023 The KubeAdmiral Authors.
+*/
+
+package v1alpha1
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:validation:Required
+// +kubebuilder:resource:path=collectedstatuses,shortName=cs,singular=collectedstatus
+// +kubebuilder:object:root=true
+
+// CollectedStatus stores the collected fields from each cluster object for a FederatedObject.
+type CollectedStatus struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Clusters []ClusterObject `json:"clusters"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+
+// CollectedStatusList contains a list of CollectedStatuses.
+type CollectedStatusList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []CollectedStatus `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:validation:Required
+// +kubebuilder:resource:path=clustercollectedstatuses,shortName=ccs,singular=clustercollectedstatus,scope=Cluster
+// +kubebuilder:object:root=true
+
+// ClusterFederatedObject stores the collected fields from each cluster object for a ClusterFederatedObject.
+type ClusterCollectedStatus struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Clusters []ClusterObject `json:"clusters"`
+	// LastUpdateTime is the last time status collection was performed.
+	LastUpdateTime metav1.Time `json:"lastUpdateTime"`
```

Comment:
Why does `ClusterCollectedStatus` have `LastUpdateTime` but `CollectedStatus` does not?

Can we make both of them use the same struct for fields?